umbraco/Umbraco-CMS
# V8 fix remove custom event directives

### Prerequisites


If there's an existing issue for this PR then this fixes: https://github.com/umbraco/Umbraco-CMS/issues/3612

### Description

This PR replaces some of the custom event directives with the native Angular equivalents.

### Summary of changes

- `onKeyup` and `onKeydown` are no longer in use. They have been removed because there are native `ng-` equivalents available ([`ngKeyUp`](https://docs.angularjs.org/api/ng/directive/ngKeyup) and [`ngKeydown`](https://docs.angularjs.org/api/ng/directive/ngKeydown)).
- `onBlur` and `onFocus` have been removed and replaced with `ngBlur` and `ngFocus`.
- The drag event directives (`onDrag*`) along with `onOutsideClick`, `onRightClick` and `onDelayedMouseleave` remain unchanged as there are no native `ng-` equivalents.
  - `onDragOver`, `onDragStart` and `onDrop` are no longer in use and could be removed. However there might be 3rd party packages depending on them, so best to err on the side of caution with those.

### Test limitations

While working on this PR I discovered a few bugs in the current codebase. I will investigate them closer and create issues/PRs for them later on, but as they have an impact on testing this PR, here's a quick run-down:

1. Embedding something (e.g. a YouTube video from the RTE) fails if:
	1. You submit an empty width or height.
	2. You change the default width or height without explicitly blurring the input fields before submitting the dialog.
2. The media picker won't let you create new folders.
3. The tags editor throws a console error whenever you add a tag.

### Test instructions

Taking the limitations above into account, here's what should be tested for this PR:

1. It's possible to change the width and height of an embedded YouTube movie in the RTE (while explicitly tabbing out of the width and height fields before submitting the dialog).
2. Given a doctype with several property groups - in the doctypes builder:
	1. Clicking a group activates the group (adds the "active" border around the group).
	2. Clicking "Add Property" on a group also activates the group.
	3. Changing sort order of the groups using the sort order input fields still works.
	4. When you click a locked field (doctype or property alias) it unlocks for editing, and it locks again when you blur the field.
